### PR TITLE
Handle PRs created by dependabot in auto-pr GHA workflow

### PR DESCRIPTION
**Pull request by komamitsu**
### Context

We've introduced https://github.com/scalar-labs/scalardb/pull/884 and related changes. But we ran into a failure https://github.com/scalar-labs/scalardb/actions/runs/5183834740/jobs/9342161161 that occurred as the original PR creator `dependabot` wasn't an actual GitHub user and the workflow failed to assign `dependabot` to the new PRs.

### Changes

This PR updates the GHA workflow and related scripts to avoid assigning someone to new PRs and issues if it can't get GitHub user information based on the original PR creator name.

### CODE REVIEW

File: .github/workflows/auto-pr.yaml [hunk 1]
```diff
@@ -23,6 +23,17 @@ jobs:
 
       - name: Create pull requests
         run: |
+          assignee=$(ci/auto-pr/fetch_gh_user_info "${{ github.event.repository.owner.login }}" "${{ github.event.repository.name }}" "${{ github.event.pull_request.user.login }}")
+          echo -------------
+          echo "assignee: $assignee"
+          echo -------------
+          if [[ -z $assignee ]]; then
+            # For instance, we can't assign `debendabot` to a new PR.
```
Comment (Torch3333) [inline .github/workflows/auto-pr.yaml:31]:
```suggestion
            # For instance, we can't assign `dependabot` to a new PR.
```

**Review by Torch3333 — APPROVED** [review]
LGTM, thank you!

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

File: ci/auto-pr/create_pull_requests [hunk 2]
```diff
@@ -32,16 +32,29 @@ function cherry_pick_and_create_pull_request () {
   git cherry-pick --no-rerere-autoupdate -m1 $commit_sha
   git push origin $new_branch
   git status
-  gh pr create --assignee $assignee \
-    --base "$branch" \
-    --title "Backport to branch($branch) : $pull_request_title" \
-    --body "Backport of $pull_request_url"
+  if [[ -z $assignee ]]; then
+    gh pr create \
```
Comment (brfrn169) [inline ci/auto-pr/create_pull_requests:36]: I'm wondering, if no one is assigned to a PR or an issue, who will notice it? One possible idea is to assign it to the person who merged the original PR. What do you think?

File: .github/workflows/auto-pr.yaml (hunk 1 above)
Comment (komamitsu) [reply .github/workflows/auto-pr.yaml]: Oops. Thanks!

File: ci/auto-pr/create_pull_requests (hunk 2 above)
Comment (komamitsu) [reply ci/auto-pr/create_pull_requests:36]: Sounds a good idea. I'll try it.

Comment (komamitsu) [reply ci/auto-pr/create_pull_requests]: @brfrn169 Fixed it in https://github.com/scalar-labs/scalardb/pull/898/commits/ea9b8b4d1785f8b056d1007814017e0b6825e255

**Review by brfrn169 — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**